lib/publiccloud/azure.pm

```diff
@@ -46,7 +46,18 @@ sub decode_azure_json {
     return decode_json(colorstrip(shift));
 }
 
-sub resource_exist {
+=head2 resource_group_exist
+
+    $self->resource_group_exist();
+
+Checks if the Azure resource group exists
+
+ * Returns undef if the resource group does not exist otherwise returns json
+
+=cut
+
+
+sub resource_group_exist {
```

Comment:
A subroutine named `resource_group_exist` should return a boolean value. I would recommend to rename this subroutine to something like `resource_group_json`.

Reply:
OK like this?
```perl
=head2 resource_group_exist

    $self->resource_group_exist();

Checks if the Azure resource group exists and returns boolean

=cut


sub resource_group_exist {
    my ($self) = @_;
    my $group = $self->resource_group;
    my $output = script_output_retry("az group show --name '$group' --output json", retry => 3, timeout => 30, delay => 10);
    return ($output ne '[]') ? 1 : 0;
}
```

Reply:
That's also an option, yes :+1: